pkg/zones/wellknown.go

```diff
@@ -389,7 +392,18 @@ func GuessCloudForZone(zone string) (kops.CloudProviderID, bool) {
 func WellKnownZonesForCloud(matchCloud kops.CloudProviderID) []string {
 	switch matchCloud {
 	case kops.CloudProviderAWS:
-		return awsZones
+		awsCloud, err := awsup.NewAWSCloud("us-east-1", map[string]string{})
+		zones, err := awsCloud.EC2().DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{
+			AllAvailabilityZones: aws.Bool(true),
```

Comment:
We possibly shouldn't set this for the command-completion case.

Reply:
Do we want to get the exact zones, or regions would be good enough and allow operators to type the final character? For AWS we could use the partitions list:
https://github.com/aws/aws-sdk-go/blob/1356aada4806053cf6a64f66288912c9625fa016/aws/endpoints/defaults.go#L103

Reply:
For the command-completion case, we're supposed to list all the possibilities so that the shell can list them out.

In the infer-the-cloud case we could be lax.

Reply:
This approach can only be used for zones. Cloud inference is not possible as you don't have credentials for all types of clouds.

Reply:
Returning the region name as the completion gives a better command-completion experience, so I changed it to that.

Reply:
You can infer clouds that you have credentials for. Creating clusters in clouds you don't have credentials for isn't going to work anyway.

Reply:
I appreciate this change. It is a very good approach for the region command-completion for AWS.
Though, looping through N clouds just to infer the cloud provider is not an optimal approach. It would most likely become slow and unpredictable the more cloud providers we add to the mix.

Reply:
For one, is there any reason this PR shouldn't go in as-is?

This AWS code is able to quickly determine, without making remote calls, that it doesn't have any matches when it is given regions from another cloud provider.